### PR TITLE
Validate axis in `CAReduce` and handle negative axis in `squeeze`

### PR DESCRIPTION
**Pull request by ricardoV94**

### CODE REVIEW

**Comment by kc611** [comment]
For the squeeze issue we can do that by merging: https://github.com/aesara-devs/aesara/pull/657
Have a quick look at and we can merge it, it's looks complete to me. 

As for the CAReduce issue I had a PR for that too, but it was more general in nature: https://github.com/aesara-devs/aesara/pull/342, That would need to rebased though. It's also complete. (Mostly)

**Comment by ricardoV94** [comment]
> For the squeeze issue we can do that by merging: #657 Have a quick look at and we can merge it, it's looks complete to me.
> 
> As for the CAReduce issue I had a PR for that too, but it was more general in nature: #342, That would need to rebased though. It's also complete. (Mostly)

I think it might still be worth pushing this PR, since it's pretty self-contained and adds regression tests for behavior that was defective until now.

**Comment by kc611** [comment]
Sure 👍, seems like the PR I tagged needs some discussion before it can be merged.

File: tests/tensor/test_extra_ops.py [hunk 1]
```diff
@@ -404,6 +404,19 @@ def test_axis(self):
 
         assert res.broadcastable == (False, False)
 
+        variable = TensorType(config.floatX, [True, False, True, False, True])()
+        res = squeeze(variable, axis=(0, -1))
+
+        assert res.broadcastable == (False, True, False)
```
Comment (kc611) [inline tests/tensor/test_extra_ops.py:410]: A little tidbit here, we should make these cases with `pytest.parametrize`

**Review by kc611 — APPROVED** [review]
Rest of it LGTM!

However down the line we should definitely be making squeeze an `Op`. `DimShuffle` in my opinion is doing two kinds of atomic operations right now. i.e adding/removing axes and shuffling them. If not in three like in #657 then atleast in two different `Op`s doing the above two kinds of operations. We can continue this discussion in #657

File: tests/tensor/test_extra_ops.py (hunk 1 above)
Comment (ricardoV94) [reply tests/tensor/test_extra_ops.py:410]: `parametrize` makes tests a bit less readable but has the advantage of being a bit quicker to spot the failing condition, and reusing code. The latter is not really a thing here though... Anyway, I would not object to refactoring it :)

File: aesara/tensor/extra_ops.py [hunk 2]
```diff
@@ -615,24 +615,34 @@ def squeeze(x, axis=None):
     x :
         Input data, tensor variable.
     axis : None or int or tuple of ints, optional
-        Selects a subset of the single-dimensional entries in the
-        shape. If an axis is selected with shape entry greater than
-        one, an error is raised.
+        Selects a subset of broadcastable dimensions to be revomed.
```
Comment (brandonwillard) [inline aesara/tensor/extra_ops.py:618]:
```suggestion
        Selects a subset of broadcastable dimensions to be removed.
```

**Review by brandonwillard — CHANGES_REQUESTED** [review]
Just some minor docstring updates; otherwise, this looks good.

**Review by brandonwillard — APPROVED** [review]

**Review by brandonwillard — APPROVED** [review]

**Pull request merged**